feat(bagel): add multimodal Bagel training support (#2275)
NVIDIA-NeMo/Automodel: docs/model-coverage/overview.md

Reviewer: ```suggestion
NeMo AutoModel integrates with Hugging Face `transformers`. Any LLM or VLM that can be instantiated through `transformers` can also be used with NeMo AutoModel, subject to runtime, third-party software dependencies, and feature compatibility.
```

Reviewer: ```suggestion
If a model from the Hub doesn't work as expected, refer to the [Troubleshooting Guide](troubleshooting.md) for common issues and solutions.
```

|------------|------|--------|---------|
| `AutoModelForCausalLM` | Text Generation (LLM) | Supported | See [LLM model list](llm/index.md). |
| `AutoModelForImageTextToText` | Image-Text-to-Text (VLM) | Supported | See [VLM model list](vlm/index.md). |
| Custom multimodal models | Unified multimodal training | Supported | See [Multimodal model list](multimodal/index.md). |
| `AutoModelForSequenceClassification` | Sequence Classification | WIP | Early support; interfaces may change. |
| Diffusers Pipelines | Diffusion Generation (T2I, T2V) | Supported | See [Diffusion model list](diffusion/index.md). |
| `NeMoAutoModelBiEncoder` | Embedding Models | Supported | See [Embedding model list](embedding/index.md). |
| `NeMoAutoModelCrossEncoder` | Reranking Models | Supported | See [Reranking model list](reranker/index.md). |

## Release Log

> The table below tracks when model support and key features were added across NeMo AutoModel releases. For the full list of tested architectures and example configs, see the [LLM](llm/index.md), [VLM](vlm/index.md), and [Multimodal](multimodal/index.md) pages.

Reviewer: ```suggestion
The table below tracks when model support and key features were added across NeMo AutoModel releases. For the full list of tested architectures and example configs, refer to the [LLM](llm/index.md), [VLM](vlm/index.md), and [Multimodal](multimodal/index.md) pages.
```
